source/FPSciApp.cpp

```diff
@@ -1159,6 +1163,43 @@ void FPSciApp::onNetwork() {
 				m_networkFrameNum = frameNum; // Set the frame number to sync with the server
 				debugPrintf("Recieved a request to start session.\n");
 			}
+			else if (type == NetworkUtils::MessageType::SEND_PLAYER_CONFIG_TO_CLIENTS) {
+				shared_ptr<PlayerEntity> player = scene()->typedEntity<PlayerEntity>("player");
+
+				*player->moveRate = packet_contents.readFloat32();
+				*player->moveScale = packet_contents.readVector2();
+
+				(*player->axisLock)[0] = packet_contents.readBool8();
+				(*player->axisLock)[1] = packet_contents.readBool8();
+				(*player->axisLock)[2] = packet_contents.readBool8();
+
+				*player->accelerationEnabled = packet_contents.readBool8();
+				*player->movementAcceleration = packet_contents.readFloat32();
+				*player->movementDeceleration = packet_contents.readFloat32();
+
+				*player->sprintMultiplier = packet_contents.readFloat32();
+
+				*player->jumpVelocity = packet_contents.readFloat32();
+				*player->jumpInterval = packet_contents.readFloat32();
+				*player->jumpTouch = packet_contents.readBool8();
+
+				*player->height = packet_contents.readFloat32();
+				*player->crouchHeight = packet_contents.readFloat32();
+
+				*player->headBobEnabled = packet_contents.readBool8();
+				*player->headBobAmplitude = packet_contents.readFloat32();
+				*player->headBobFrequency = packet_contents.readFloat32();
+
+				*player->respawnPos = packet_contents.readVector3();
+				*player->respawnToPos = packet_contents.readBool8();
+
+				*player->movementRestrictionX = packet_contents.readFloat32();
+				*player->movementRestrictionZ = packet_contents.readFloat32();
+				*player->restrictedMovementEnabled = packet_contents.readBool8();
+
+				*player->counterStrafing = packet_contents.readBool8();
+				debugPrintf("Recieved a request to set player config.\n");
+			}
```

Comment:
I would think carefully about how this transport is being done. By setting the `PlayerEntity` parameters directly but not overriding the config that is used to set them in `FPSciApp::initPlayer()` you risk potentially losing this config whenever a method like `FPSciApp::updateSession()` is called. I think it would best to update the local `PlayerConfig` so that correct values are logged and updated.

Reply:
Updated this to write directly to the session - playerConfig.